sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2085,37 +2084,9 @@ class Dataset[T] private[sql](
       "in the right attributes",
```

Comment:
shall we move the check to the analyzer rule as well?

Reply:
ok. moved.